### ISSUE TITLE
Enabling softcap option

### ISSUE THREAD

**Issue by SpyrosMouselinos** [facebookresearch/xformers#1113]
# 🚀 Feature

I would like to have a softcap argument available in the memory_efficient_attention function. Flash-Attention 2 supports this but is internally set to 0.0 in the xformers library.

## Motivation

Soft capping prevents logits from growing excessively large without truncating them, and it has been shown to help with the stability of LLMs in the range of tens of billions of parameters, as demonstrated in Gemma 2 (https://arxiv.org/abs/2408.00118).
The mechanism is pretty simple and is applied to the pre-attention logits as follows:

logits ← soft_cap ∗ tanh(logits/soft_cap), before being passed to the softmax layer

## Pitch

I would like to have softcap available as an argument in the memory_efficient_attention function, it would be a float argument with strictly positive values.

**Comment by lw** [comment]
We don't expect to expose this feature via xFormers, mainly because it is relatively easily achievable via PyTorch's new FlexAttention mechanism, which is designed for this kind of variations over self-attention.

You can read more about it here: https://pytorch.org/blog/flexattention/. You can even find a softcap example.
